### PR TITLE
IIO driver for AD74413R

### PR DESCRIPTION
**Pull request by CiprianRegus**
Implemented the IIO driver and example project for the AD74413R.

### CODE REVIEW

File: drivers/adc-dac/ad74413r/iio_ad74413r.c [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+			case AD74413R_ADC_RANGE_2P5V_INT_POW:
+				val = -AD74413R_ADC_MAX_VALUE;
+				break;
+			case AD74413R_ADC_RANGE_5V_BI_DIR:
+				val = -(AD74413R_ADC_MAX_VALUE / 2);
+				break;
+			default:
+				return -EINVAL;
+			}
+
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.c:317]: nit: extra line

File: drivers/adc-dac/ad74413r/iio_ad74413r.c [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+}
+
+/**
+ * @brief Stop the ADC conversions.
+ * @param dev - The iio device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+static int ad74413r_iio_buffer_disable(void *dev)
+{
+	struct ad74413r_iio_desc *iio_desc = dev;
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.c:615]: wondering if we can avoid here this extra variable, and just type cast in the function call :)

File: drivers/adc-dac/ad74413r/iio_ad74413r.c [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+	int ret;
+	uint32_t j = 0;
+	uint32_t val;
+	struct ad74413r_iio_desc *iio_desc;
+	struct ad74413r_desc *ad74413r_desc;
+
+	iio_desc = dev;
+	ad74413r_desc = iio_desc->ad74413r_desc;
+
+	for (uint32_t i = 0; i < samples; i++) {
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.c:639]: we should declare the type of the variable at the start of the function (according to the linux drivers code guidelines). Same for `chan_i`. check the entire code for this type of nits

File: drivers/adc-dac/ad74413r/iio_ad74413r.c [hunk 4]
```diff
@@ -0,0 +1,763 @@
...
+	uint32_t j = 0;
+	uint32_t val;
+	uint32_t buff[4];
+	struct ad74413r_iio_desc *iio_desc;
+	struct ad74413r_desc *desc;
+
+	iio_desc = dev_data->dev;
+	desc = iio_desc->ad74413r_desc;
+
+	for (uint32_t i = 0; i < AD74413R_N_CHANNELS; i++) {
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.c:671]: ditto

File: drivers/adc-dac/ad74413r/iio_ad74413r.c [hunk 5]
```diff
@@ -0,0 +1,763 @@
...
+			    init_param->ad74413r_init_param);
+	if (ret)
+		goto err;
+
+	ret = ad74413r_clear_errors(descriptor->ad74413r_desc);
+	if (ret)
+		goto err;
+
+	/** The operation modes for the physical channels are set only at initialization. */
+	for (uint32_t i = 0; i < AD74413R_N_CHANNELS; i++) {
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.c:715]: ditto

File: drivers/adc-dac/ad74413r/iio_ad74413r.h [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+#ifndef IIO_AD74413R_H
+#define IIO_AD74413R_H
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "iio.h"
+#include "ad74413r.h"
+
+
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.h:48]: nit: extra lines

File: projects/ad74413r/src/platform/maxim/parameters.h [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /***************************************************************************//**
  *   @file   parameters.h
- *   @brief  Definitions specific to Maxim platform used by eval-adxl355-pmdz
```
Comment (amiclaus) [inline projects/ad74413r/src/platform/maxim/parameters.h:3]: this looks like an unrelated change :)

**Review by amiclaus — COMMENTED** [review]
some comments on my side, otherwise lgtm.

**Comment by CiprianRegus** [comment]
Implemented review findings.

**Comment by CiprianRegus** [comment]
Implemented review findings.

File: drivers/adc-dac/ad74413r/iio_ad74413r.c [hunk 8]
```diff
@@ -0,0 +1,751 @@
...
+	uint32_t n_chan = 0;
+	uint32_t channel_buff_cnt = 0;
+	struct ad74413r_desc *ad74413r_desc;
+	struct ad74413r_channel_config *config;
+	struct iio_channel *chan;
+	struct iio_channel *chan_buffer;
+	struct ad74413r_channel_map channels_info;
+
+	ad74413r_desc = iio_desc->ad74413r_desc;
+	config =  ad74413r_desc->channel_configs;
```
Comment (amiclaus) [inline drivers/adc-dac/ad74413r/iio_ad74413r.c:533]: nit: extra space :)

**Review by amiclaus — APPROVED** [review]
one nit, the rest lgtm :)

**Comment by CiprianRegus** [comment]
Removed extra space.
